sklearn/gaussian_process/_gpr.py

```diff
@@ -196,10 +196,16 @@ def fit(self, X, y):
         if self.normalize_y:
             self._y_train_mean = np.mean(y, axis=0)
             self._y_train_std = np.std(y, axis=0)
-
+            # assign _y_train.std to 1 when _y_train.std is zero
+            # to avoid divide by zero error
+            if np.iterable(self._y_train_std):
+                self._y_train_std = np.asarray(
+                        [std if std else 1 for std in self._y_train_std])
+            else:
+                self._y_train_std = \
+                    self._y_train_std if self._y_train_std else 1
```

Comment:
Although it would work, I think that for clarity we should use:

```suggestion
                self._y_train_std = (
                    self._y_train_std if self._y_train_std == 0 else 1)
```